### PR TITLE
Hot fixes to unit tests

### PR DESCRIPTION
**Pull request by rjbautis**
* Cleans up test cases
* Removes and/or replaces broken DOS URIs in certain unit test cases

### CODE REVIEW

**Comment by denis-yuen** [comment]
Looks like certain build dependencies are missing, @coverbeck vaguely recall an issue having to do with this. How did this work before?

**Comment by coverbeck** [comment]
@denis-yuen Not sure, will investigate. Also, I think this PR is not quite ready and should not yet have been assigned reviewers; sorry for the confusion.

File: src/main/java/io/dockstore/provision/DOSPlugin.java
Comment (coverbeck) [inline src/main/java/io/dockstore/provision/DOSPlugin.java:63]: This is really a comment for line 55, but GitHub doesn't let a user comment on lines that weren't changed...

This is really minor, but I would have a static final for `return new HashSet..`. Otherwise you're creating new objects every time this method is invoked.

In this particular case it won't effectively matter, as the method is only called twice and 2 objects will have no noticeable impact on performance, but just mentioning as a learning experience for future programming where a similar method might get invoked tens of thousands of times.

File: src/main/java/io/dockstore/provision/DOSPluginUtil.java
Comment (coverbeck) [inline src/main/java/io/dockstore/provision/DOSPluginUtil.java]: I think `new JSONObject(content)` will fail if content is null. Are you relying on the catch to handle that case? I think it's clearer if you do an explicit null check. One way could be:

```
if (content != null) {
   return Optional.of(new JSONObject(content))
}
```

Or you could do something fancier with a map

File: src/main/java/io/dockstore/provision/DOSPluginUtil.java
Comment (coverbeck) [inline src/main/java/io/dockstore/provision/DOSPluginUtil.java]: Java variable names should not start with a capital, unless they are static final, in which case they are all caps.

Comment (coverbeck) [reply src/main/java/io/dockstore/provision/DOSPluginUtil.java]: Also, I would move both this and the preceding declaration inside the try block and initialize/declare them together, e.g.,

```
final JSONResponse jsonResponse = downloadJSON(immutableTriple);
```

File: src/main/java/io/dockstore/provision/DOSPluginUtil.java
Comment (coverbeck) [inline src/main/java/io/dockstore/provision/DOSPluginUtil.java]: It would be better if this were a method variable rather than a field, IMHO.

I guess it's tricky to close the connection though.

**Review by coverbeck — CHANGES_REQUESTED** [review]
A lot of nit-picky stuff, but this is also (hopefully) a learning/teaching experience.

Also, as we discussed, I would not call this "Hot fixes".

Finally, I think your commit message should both reference the GitHub issue, and it should mention that you are removing dependencies on external servers for the tests to pass.

File: src/main/java/io/dockstore/provision/DOSPlugin.java
Comment (rjbautis) [reply src/main/java/io/dockstore/provision/DOSPlugin.java:63]: Sounds good, I will include the change.

**Review by coverbeck — APPROVED** [review]
Looks good!

Please wait for approval from @garyluu and @agduncan94 before merging...

**Review by agduncan94 — APPROVED** [review]

**Review by garyluu — APPROVED** [review]

**Comment by rjbautis** [comment]
* Updates unit tests to remove dependency on external servers using proper mocks
* Refactors method names & signatures as well as variables in src code

**Pull request merged**
